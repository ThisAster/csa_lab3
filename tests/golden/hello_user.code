[{"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "immediate", "arg": 2},
 {"opcode": "st", "arg_type": "address", "arg": 1},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "sub", "arg_type": "immediate", "arg": 18},
 {"opcode": "jl", "arg_type": "address", "arg": 9},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 10},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "je", "arg_type": "address", "arg": 22},
 {"opcode": "ld", "arg_type": "address", "arg": 1},
 {"opcode": "st", "arg_type": "address", "arg": 21},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "address", "arg": 21},
 {"opcode": "st", "arg_type": "address", "arg": 22},
 {"opcode": "ld", "arg_type": "indirect_address", "arg": 22},
 {"opcode": "out", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "immediate", "arg": 1},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 4},
 {"opcode": "ld", "arg_type": "immediate", "arg": 24},
 {"opcode": "st", "arg_type": "address", "arg": 23},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "st", "arg_type": "address", "arg": 125},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "sub", "arg_type": "immediate", "arg": 100},
 {"opcode": "jl", "arg_type": "address", "arg": 33},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 34},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "je", "arg_type": "address", "arg": 73},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "address", "arg": 23},
 {"opcode": "st", "arg_type": "address", "arg": 126},
 {"opcode": "in", "arg_type": "address", "arg": 0},
 {"opcode": "st", "arg_type": "indirect_address", "arg": 126},
 {"opcode": "ld", "arg_type": "address", "arg": 23},
 {"opcode": "st", "arg_type": "address", "arg": 127},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "address", "arg": 127},
 {"opcode": "st", "arg_type": "address", "arg": 128},
 {"opcode": "ld", "arg_type": "indirect_address", "arg": 128},
 {"opcode": "sub", "arg_type": "immediate", "arg": 10},
 {"opcode": "je", "arg_type": "address", "arg": 50},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 51},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "je", "arg_type": "address", "arg": 54},
 {"opcode": "ld", "arg_type": "immediate", "arg": 100},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "address", "arg": 23},
 {"opcode": "st", "arg_type": "address", "arg": 129},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "address", "arg": 129},
 {"opcode": "st", "arg_type": "address", "arg": 130},
 {"opcode": "ld", "arg_type": "indirect_address", "arg": 130},
 {"opcode": "sub", "arg_type": "immediate", "arg": 0},
 {"opcode": "jne", "arg_type": "address", "arg": 64},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 65},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "je", "arg_type": "address", "arg": 69},
 {"opcode": "ld", "arg_type": "address", "arg": 125},
 {"opcode": "add", "arg_type": "immediate", "arg": 1},
 {"opcode": "st", "arg_type": "address", "arg": 125},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "immediate", "arg": 1},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 28},
 {"opcode": "ld", "arg_type": "immediate", "arg": 132},
 {"opcode": "st", "arg_type": "address", "arg": 131},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "sub", "arg_type": "immediate", "arg": 7},
 {"opcode": "jl", "arg_type": "address", "arg": 82},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 83},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "je", "arg_type": "address", "arg": 95},
 {"opcode": "ld", "arg_type": "address", "arg": 131},
 {"opcode": "st", "arg_type": "address", "arg": 140},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "address", "arg": 140},
 {"opcode": "st", "arg_type": "address", "arg": 141},
 {"opcode": "ld", "arg_type": "indirect_address", "arg": 141},
 {"opcode": "out", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "immediate", "arg": 1},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 77},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "sub", "arg_type": "address", "arg": 125},
 {"opcode": "jl", "arg_type": "address", "arg": 102},
 {"opcode": "ld", "arg_type": "immediate", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 103},
 {"opcode": "ld", "arg_type": "immediate", "arg": 1},
 {"opcode": "je", "arg_type": "address", "arg": 115},
 {"opcode": "ld", "arg_type": "address", "arg": 23},
 {"opcode": "st", "arg_type": "address", "arg": 142},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "address", "arg": 142},
 {"opcode": "st", "arg_type": "address", "arg": 143},
 {"opcode": "ld", "arg_type": "indirect_address", "arg": 143},
 {"opcode": "out", "arg_type": "address", "arg": 0},
 {"opcode": "ld", "arg_type": "address", "arg": 0},
 {"opcode": "add", "arg_type": "immediate", "arg": 1},
 {"opcode": "st", "arg_type": "address", "arg": 0},
 {"opcode": "jmp", "arg_type": "address", "arg": 97},
 {"opcode": "ld", "arg_type": "immediate", "arg": 33},
 {"opcode": "out", "arg_type": "address", "arg": 0},
 {"opcode": "hlt", "arg_type": "none", "arg": 0}]