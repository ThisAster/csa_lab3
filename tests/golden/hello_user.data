[0, 0, 87, 104, 97, 116, 32, 105, 115, 32, 121, 111, 117, 114, 32, 110, 97, 109, 101, 63, 0, 0, 0, 0, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 95, 0, 0, 0, 0, 0, 0, 0, 0, 72, 101, 108, 108, 111, 44, 32, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]