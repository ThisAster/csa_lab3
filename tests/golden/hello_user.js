var i
i = 0
var greating
greating = "What is your name?"
while (i < 18) {
    print(greating[i])
    i = i + 1
}
var username
username = "____________________________________________________________________________________________________"
var n
n = 0
i = 0
while (i < 100) {
    username[i] = input()
    if (username[i] == 10) {
        i = 100
    }
    if (username[i] != 0) {
        n = n + 1
    }
    i = i + 1
}
var s
s = "Hello, "
i = 0
while (i < 7) {
    print(s[i])
    i = i + 1
}
i = 0
while (i < n) {
    print(username[i])
    i = i + 1
}
print(33)
