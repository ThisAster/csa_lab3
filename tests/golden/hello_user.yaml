js: hello_user.js
code: hello_user.code
data: hello_user.data
txt: hello_user.txt
translator_log: hello_user-1.log
machine_log: hello_user-2.log
